src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -385,6 +385,34 @@ private ReadOnlySequence<T> SliceImpl(in SequencePosition begin, in SequencePosi
             );
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private SequenceType GetSequenceType()
+        {
+            // We take high order bits of two indexes and move them
+            // to a first and second position to convert to SequenceType
+
+            // if (start < 0  and end < 0)
+            // start >> 31 = -1, end >> 31 = -1
+            // 2 * (-1) + (-1) = -3, result = (SequenceType)3
+
+            // if (start < 0  and end >= 0)
+            // start >> 31 = -1, end >> 31 = 0
+            // 2 * (-1) + 0 = -2, result = (SequenceType)2
+
+            // if (start >= 0  and end >= 0)
+            // start >> 31 = 0, end >> 31 = 0
+            // 2 * 0 + 0 = 0, result = (SequenceType)0
+
+            // if (start >= 0  and end < 0)
+            // start >> 31 = 0, end >> 31 = -1
+            // 2 * 0 + (-1) = -1, result = (SequenceType)1
+
+            return (SequenceType)(-(2 * (_sequenceStart.GetInteger() >> 31) + (_sequenceEnd.GetInteger() >> 31)));
```

Comment:
would it make sense to do unchecked cast to uint and get rid of negation and simplify the comment?

Reply:
Yes. I will consider this change separately (along with casting to uint in other places we call `GetInteger()`). Thanks.